src/probnum/randprocs/_gauss_markov_process.py

```diff
@@ -0,0 +1,75 @@
+"""Gauss-Markov Processes."""
+from typing import Union
+
+import numpy as np
+
+from probnum import randvars, statespace
+from probnum.type import FloatArgType
+
+from ._gaussian_process import GaussianProcess
+
+_InputType = Union[np.floating, np.ndarray]
+_OutputType = Union[np.floating, np.ndarray]
+
+
+class GaussMarkovProcess(GaussianProcess):
+    r"""Gaussian processes with the Markov property.
+
+    A Gauss-Markov process is a Gaussian process with the additional property that
+    conditioned on the present state of the system its future and past states are
+    independent. This is known as the Markov property or as the process being
+    memoryless. Gauss-Markov processes :math:`x_t` can be defined as the solution of a
+    (linear) transition model given by a stochastic differential equation (SDE) of
+    the form
+
+    .. math:: \mathrm{d} x_t = G(t) x_t \mathrm{d} t + \mathrm{d} w_t.
+
+    and a Gaussian initial condition :math:`x_0`.
+
+    Parameters
+    ----------
+    linear_sde
+        Linear stochastic differential equation describing a state transition of the
+        system.
+    t0
+        Initial starting index / time of the process.
+    x0
+        Gaussian random variable describing the initial state.
+
+    See Also
+    --------
+    RandomProcess : Random processes.
+    GaussianProcess : Gaussian processes.
+
+    Examples
+    --------
+    """
+    # pylint: disable=invalid-name
+
+    def __init__(
+        self,
+        linear_sde: statespace.LinearSDE,
+        x0: randvars.Normal,
+        t0: FloatArgType = 0.0,
+    ):
+        self.linear_sde = linear_sde
+        self.t0 = t0
+        self.x0 = np.asarray(x0).reshape(1, -1)
+
+        super().__init__(
+            input_dim=1,
+            output_dim=1 if self.x0.ndim == 0 else self.x0.shape[0],
+            mean=lambda t: self.__call__(t).mean,
+            cov=lambda t: self.__call__(t).cov,
+        )
+
+    def __call__(self, x: _InputType) -> randvars.Normal:
```

Comment:
While I understand that `x` is the natural input choice for a general random process, it makes me feel uncomfortable to call a Markov-process with `markov_process(x=0.1)` instead of `t`... 
I don't have an idea how to solve this problem though. Either, we could knowingly deviate from this interface in subclasses (and use `t` wherever appropriate), or change `x` to something like `point/location/input/value/...` which could be more general, or suck it up and use `x`. I dont know what to do. What do you think? 
In any case, this relates to the `x0` comment in `GaussMarkovProcess.__init__`.

Reply:
Settled on `args`. Similar naming is used in JAX.